src/main/scala/fastq.scala

```diff
@@ -269,7 +269,7 @@ case object fastq {
 
     def parseFastqPhred33: Iterator[Option[FASTQ]] =
       lines
-        .grouped(4)
+        .grouped(4).filter(_.size == 4) // grouped is broken
```

Comment:
That's not another pass. And it is broken, check http://stackoverflow.com/a/7959133/614394 and http://www.scala-lang.org/old/node/7939